### PR TITLE
Librairies.io

### LINKED ISSUE #55: Collect metadata from public sector libraries
We started to list public sector [libraries](https://github.com/DISIC/politique-de-contribution-open-source/blob/master/bibliotheques-organismes-publics).  We don't display anything about them on https://code.etalab.gouv.fr so far, but we want to do so.

The challenge is this: given an entry (a line) in the text file mentioned above, how to retrieve the complete list of packages that are published by an account, on all library platforms (npm, maven, etc.)?

### PR DESCRIPTION
**Pull request by guillim**
# Why
fix #55 

## What
We are submitting a first way to get info on packages owned and maintained by the french gov.

## How
Using the API Libraries.io

## TODO
- Probably exploding the limits of the API key, we should consider having some premium account. Note that on their website, only a limitation rate exists, letting me think there are no request limitation even in the free account.
- set up a 60/request/minute limitation in our calls [at the moment, we have 429 answer when exceeding the limit]

### CODE REVIEW

**Comment by guillim** [comment]
About the limitation rate, I implemented a first way : I artificially slow down the python process (making it sleep 1 second every time it sends a request to libraries.io)

This should do the trick at the moment.

Another option to increase the speed would be to get 2 API keys ... but probably against the **"Terms and Conditions"** of the website

File: fetch.py [hunk 1]
```diff
@@ -52,4 +52,13 @@ def fetch_orgs(detector):
         all_orgs[k].append(v)
 
 save_orgs(all_orgs)
+
+# Save packages created by each org
+all_packs = []
+
+print("Fetching librairies.io for packages")
```
Comment (bzg) [inline fetch.py:59]: Nitpick: "Fetching packages from libraries.io" sounds more proper.

File: models.py [hunk 2]
```diff
@@ -47,3 +47,25 @@ class Organisation(BaseModel):
     nombre_repertoires: int
     date_creation: str
     plateforme: str
+
+@dataclass(frozen=True)
+class Package(BaseModel):
+    deprecation_reason: bool
```
Comment (bzg) [inline models.py:53]: I haven't closely checked about deprecation_reason being a bool, I assume you did, please ignore this if so.

File: utils.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import os
+import requests, json, time
+from collections import defaultdict
+
+from models import Package
+
+def fetch_package(org):
+    api_key = os.getenv("LIBRAIRIES_API_KEY")
```
Comment (bzg) [inline utils.py:8]: I suggest replacing LIBRAIRIES_API_KEY by LIBRARIES.IO_API_KEY (at least it should be LIBRARIES, not LIBRAIRIES.)

**Review by bzg — CHANGES_REQUESTED** [review]
Thanks a lot!  LGTM.  The PR also needs to edit https://github.com/etalab/data-codes-sources-fr/blob/master/.circleci/config.yml#L35 to remove `data/packages/*`.

PS: The mix of french and english in the directory structure is not a problem for this PR, we can fix it while taking care of https://github.com/etalab/data-codes-sources-fr/issues/54.

File: models.py (hunk 2 above)
Comment (guillim) [reply models.py:53]: I have not yet actually. I took this schema as a dumb dev :)

**Comment by guillim** [comment]
> Also, CircleCI should know about the `LIBRARIES.IO_API_KEY` environment variable.

I don't have access to CircleCI, do I ?

**Comment by bzg** [comment]
Since you're a member of the Etalab organization, you should be able to connect to https://circleci.com with your GitHub account and see Etalab's projects there. Let me know if it does not work!

**Comment by guillim** [comment]
> Since you're a member of the Etalab organization, you should be able to connect to https://circleci.com with your GitHub account and see Etalab's projects there. Let me know if it does not work!

true => ✅ 

Also I had to modify the name again since "." are not allowed in env variables from the CircleCI interface

**Comment by bzg** [comment]
> Also I had to modify the name again since "." are not allowed in env variables from the CircleCI interface

Er... my bad, I suscepted something like that could happen :)

**Review by bzg — APPROVED** [review]

**Pull request merged**